external-crates/move/move-compiler/src/naming/translate.rs

```diff
@@ -1411,6 +1461,39 @@ fn exp_(context: &mut Context, e: E::Exp) -> N::Exp {
                 }
             }
         }
+        EE::Call(ma, false, tys_opt, rhs) if context.is_struct_reference(&ma) => {
+            context
+                .env
+                .check_feature(FeatureGate::PositionalFields, context.current_package, eloc);
+            let nes = call_args(context, rhs);
+            match context.resolve_struct_name(eloc, "construction", ma, tys_opt) {
+                None => {
+                    assert!(context.env.has_errors());
+                    NE::UnresolvedError
+                }
+                Some((m, sn, tys_opt, is_positional)) => {
+                    if !is_positional {
+                        let msg = "Invalid struct constructor call. Named struct declarations \
+                                                             require named constructor calls.";
+                        context
+                            .env
+                            .add_diag(diag!(NameResolution::PositionalCallMismatch, (eloc, msg)));
+                    }
+                    NE::Pack(
+                        m,
+                        sn,
+                        tys_opt,
+                        UniqueMap::maybe_from_iter(nes.value.into_iter().enumerate().map(
+                            |(idx, e)| {
+                                let field = Field::add_loc(e.loc, format!("{idx}").into());
```

Comment:
Maybe we should have a function for this. Something like `positional_field_name(idx: usize): String { format!("{idx}") }`

Just to make sure we can easily change this later if needed